dot/core/messages.go

```diff
@@ -6,15 +6,57 @@ package core
 import (
 	"errors"
 
-	"github.com/libp2p/go-libp2p-core/peer"
-
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/peerset"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/runtime"
 	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/libp2p/go-libp2p-core/peer"
 )
 
+func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt runtime.Instance,
+	tx types.Extrinsic, toPropagate *[]types.Extrinsic) (bool, error) {
```

Comment:
Use named returns, I guess the bool could be `failed`? See confusion I got below 😄 Also maybe flip it around, usually `true` means ok and `false` means not ok

```suggestion
	tx types.Extrinsic, toPropagate *[]types.Extrinsic) (failed bool, err error) {
```